### DISCUSSION TITLE
PL/SQL variable declare parsing failing with 4.6 version

### DISCUSSION THREAD

**Discussion by neeraj91github** [JSQLParser/JSqlParser#1848] [General]
Hi, I wanted to declare two variables for my query which runs on SQLServer - 

```
DECLARE @SLV_JOB_ID AS int;
DECLARE @MODEL_ID AS int;

SET @MODEL_ID = (SELECT MODEL_ID  FROM MODEL model WHERE model.NAME = ?);
SET @SLV_JOB_ID = (SELECT SLV_JOB_ID  FROM SLV_JOB job WHERE job.NAME = ?);
```

And when I parse this like this - 

`Statements statements = CCJSqlParserUtil.parseStatements(sqlStatement);`

It parses successfully. Now I need to write a similar query which will be running on Oracle DB, so I had to change the variable declaration code since variable declaration syntax is different for Oracle (PL/SQL) and SQLServer (T-SQL) - 

```
DECLARE
    g_SLV_JOB_ID NUMBER;
    g_MODEL_ID NUMBER;
BEGIN;
    SELECT MODEL_ID INTO g_MODEL_ID FROM MODEL WHERE NAME = ?;
    SELECT SLV_JOB_ID INTO g_SLV_JOB_ID FROM SLV_JOB WHERE NAME = ?;
END;
```

But parsing of this fails with the error - 

Was expecting one of:

    "@"
    "@@"
    
How to parse the PL/SQL syntax correctly ? How do I define these variables for my OracleDB script?

**Comment by manticore-projects** [comment]
Greetings.

We focus much on Standard SQL Query and DML, with some support for DDL.
Oracle PL/SQL support is scarce and you would need to provide or sponsor PRs to make it happen.
